src/Symfony/Component/DependencyInjection/Compiler/PassConfig.php

```diff
@@ -61,6 +61,7 @@ public function __construct()
             new AutowireRequiredMethodsPass(),
             new ResolveBindingsPass(),
             new AutowirePass(false),
+            new ResolveTaggedIteratorArgumentPass(),
```

Comment:
this should be before the DecoratorServicePass, so that tagging a service and then decorating it creates the reference with the original id (and so gets the decorated service rather than the inner one). It should be at the same place than the ServiceLocatorTagPass (which is here for the same reason)

Reply:
@stof debugged this real quick, and AFAIK for a decorated service we already inherit tags via `ResolveChildDefinitionsPass`, which runs sooner.

Thus `ContainerBuilder::findTaggedServiceIds()` returns expected service id's already. Am i missing something?

Reply:
child definitions are not about decorated services. Child definitions are about service definition inheritance (which is precisely why we renamed DecoratorDefinition to ChildDefinition as the previous name was totally misleading, especially after we implemented service decoration)